idft.c

```diff
@@ -0,0 +1,97 @@
+/*
+ * Copyright (C) 2022 Dominik Chat
+ *
+ * This file is part of Waves.
+ *
+ * Waves is free software: you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation, either version 3
+ * of the License, or (at your option) any later version.
+ *
+ * Waves is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ * See the GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Waves. If not, see <https://www.gnu.org/licenses/>.
+ */
+
+#include <errno.h>
+#include <stdlib.h>
+#include <string.h>
+#include <math.h>
+#include "idft.h"
+
+
+#define PI 		3.14159265
+#define TIME_DELTA	0.0001
+#define TIME_RES	0.05
+
+
+struct idft_ctx {
+	struct magnitude *mags;
+	size_t mag_cnt;
+	double time;
+};
+
+
+static double calcPoint(struct idft_ctx *ctx, double t)
+{
+	size_t i;
+	double temp;
```

Comment:
Consider renaming this variable because now is potentially misleading/confusing -
someone can understand this shortcut as temperature, and someone else as temporary.

Reply:
I do not agree - the temperature is not used in the code at all, so it will not be confused with anything.